### PR TITLE
feat(filebrowser): copy absolute path instead of relative path

### LINKED ISSUE #9485: Make Copy Path more useful
### Problem

The file browser context menu currently offers a *Copy Path* command which copies the path to the item relative to the file browser's root dir. This is not really helpful when trying to copy a path to paste into a file located elsewhere than in the file browser's root. Consider a layout like this:

```
.
├── foo
│  └── bar.ipynb
└── baz.txt
```

If I'm working on `bar.ipynb` and I copy the path to `baz.txt`, it will be just `baz.txt`, which is not appropriate for pasting into `bar.ipynb`.

### Proposed Solution

Could *Copy Path* copy the absolute file path to the item instead of the relative one? Alternatively, there could be an additional *Copy Absolute Path* command if there's a reason to keep the original command around. (I personally would rather not, it's a footgun for beginners.)

A more complicated solution would be for *Copy Path* to yield a path relative not to the file browser's root, but relative to the directory of the file currently being edited.

### Additional context

#4074, which originally requested the *Copy Path* feature, [mentioned both relative and absolute paths as options in the OP](https://github.com/jupyterlab/jupyterlab/issues/4074#issue-301350293), and one of the commenters expressed their [support for absolute paths](https://github.com/jupyterlab/jupyterlab/issues/4074#issuecomment-371573947). No one argued specifically for relative paths, though admittedly not many people weighed in.

### PR DESCRIPTION
**Pull request by pauky**
## References

Fixes #9485 
## Code changes

* add serverRoot to serverconnection in services
* add osPath to contents in services
* replace path to osPath in copy command of filebrowser-extension

## User-facing changes

None
## Backwards-incompatible changes

* Obtain absolute paths instead of relative paths when copying paths in filebrowser.

### CODE REVIEW

**Comment by pauky** [comment]
> Thanks for looking into this @pauky
> 
> This is tricky as a generic solution is gonna be though (server root is not always reliable).
> 
> Did you check if the `serverPath` is defined with the correct path?
> 
> https://github.com/jupyterlab/jupyterlab/blob/5d6db82cad747dd8384a2c3bdfb93ee85144373a/packages/services/src/contents/index.ts#L58

We can't check serverPath whether is defined with the correct path in jupyterlab. 
But I think we can only get the OS path through the jupyter_server. @fcollonval

**Comment by fcollonval** [comment]
Sorry for the late answer @pauky 

So I think the path forward should be:
- Don't add an `osPath` to the content model - we only want to focus on changing the copy path behavior
- Add a server config parameter to allow or forbid the copy of the real path as for security reasons this should not be accessible depending how the Jupyter Server is deployed.
- Condition which path is return depending on the value of the server config.

you can find an example to add a new server config parameter there: https://github.com/jupyterlab/jupyterlab/pull/12260 (it will required a PR against https://github.com/jupyterlab/jupterlab_server)

**Comment by pauky** [comment]
> 



> Sorry for the late answer @pauky
> 
> So I think the path forward should be:
> 
> * Don't add an `osPath` to the content model - we only want to focus on changing the copy path behavior
> * Add a server config parameter to allow or forbid the copy of the real path as for security reasons this should not be accessible depending how the Jupyter Server is deployed.
> * Condition which path is return depending on the value of the server config.
> 
> you can find an example to add a new server config parameter there: #12260 (it will required a PR against https://github.com/jupyterlab/jupterlab_server)

What do you mean, Jupyterab not need to adjust the parameter that controls whether the path returned by the jupyter_server is an absolute or relative path?

File: packages/filebrowser-extension/src/index.ts [hunk 1]
```diff
@@ -1209,7 +1209,7 @@ function addCommands(
         return;
       }
 
-      Clipboard.copyToSystem(item.value.path);
+      Clipboard.copyToSystem(item.value.osPath);
```
[the remark this answers is not in the document]
Comment (fcollonval) [reply packages/filebrowser-extension/src/index.ts:1212]: cc @pauky

**Comment by pauky** [comment]
@fcollonval  I recreate a PR https://github.com/jupyterlab/jupyterlab/pull/14842, according to your comment above.

**Pull request closed without merging**